samples/EventImporter/Consumer/EventsWorker.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Consumer.Models;
+
+namespace Consumer
+{
+    public class EventsWorker
+    {
+        private readonly IEventProcessor _eventHandler;
+
+        public EventsWorker(IEventProcessor eventHandler)
+        {
+            _eventHandler = eventHandler;
+        }
+
+        public void ProcessMessages(List<Event> events)
+        {
+            if (events == null)
+            {
+                throw new ArgumentNullException(nameof(events));
+            }
+
+            var operationSuccessful = events.Aggregate(true, (current, @event) => current & _eventHandler.ProcessEvent(@event));
+
+            if (!operationSuccessful)
+            {
+                throw new ProcessEventException();
+            }
+        }
+
+        public void DispatchEvent(Event eventSingle)
```

Comment:
This example is a bit confusing - why would it have separate methods for processing single or multiple events? In a real application, wouldn't you just call the List version but with only a single element inside?